### DISCUSSION TITLE
Can you prevent .eslintrc rule merging with cascading and hierarchys?

### DISCUSSION THREAD

**Discussion by Jakobud** [eslint/eslint#14228] [Q&A]
I have the following type of setup (in reality its more complicated but this example simplifies for clarity):

```
.
├── .eslintrc.json
├── js/
│   └── main.js
└── tests/
    ├── .eslintrc.json
    └── test.js
```

So `root/.eslintrc.json` has a lot of rules in it. `tests/.eslintrc.json` has minimal rules. My problem is that the `tests` js files are failing a bunch of rules that are only set in the `root/.eslintrc.json`.

When reading the documentation:

> If there is a .eslintrc file in the same directory as the file being linted, then that configuration takes precedence. ESLint then searches up the directory structure, merging any .eslintrc files it finds along the way until reaching either a .eslintrc file with root: true or the root directory.

So if I'm understanding this correctly, my `tests/` js files will use the rules in `tests/.eslintrc.json` and then also merge in the rules from `root/.eslintrc.json` (with `tests/.eslintrc.json` rules taking priority). Is this correct?

Is it possible to turn off this "merging" behavior. I want my `tests` js files to use ONLY the `tests/.eslintrc.json` file. I don't want it to cascade up the chain and merge in other `.eslintrc` configuration files.

I know that I could normally change the `tests/.eslintrc.json` rules so that the `tests` js files don't fail linting or add the `root` option, but I can't because in my situation the `tests` directory is part of a separately maintained code repo. I cannot simply make changes to it without affecting other related projects. So changing it is undesirable.

Another option is that I could move the `root/.eslintrc.json` into the `js/` directory but this is also undesirable because in reality (not in the above simplified example) I have many other directories besides the `js/` directory and they all need to use the same `.eslintrc.json`

So anyways, can I remove the merging of the rules with this setup somehow? I can see how merging of rules would be desirable in most situations. But in this particular case it is undesirable.

**Comment by Jakobud** [comment]
Ah I figured it out. Thank you`ignorePatterns`!!

**Comment by mdjermanovic** [comment]
> So if I'm understanding this correctly, my `tests/` js files will use the rules in `tests/.eslintrc.json` and then also merge in the rules from `root/.eslintrc.json` (with `tests/.eslintrc.json` rules taking priority). Is this correct?
> 
> Is it possible to turn off this "merging" behavior. I want my `tests` js files to use ONLY the `tests/.eslintrc.json` file. I don't want it to cascade up the chain and merge in other `.eslintrc` configuration files.

This would be a case for `root: true` in `tests/.eslintrc.json`.

If you can't change the `tests/.eslintrc.json` file, an alternative could be to run linting on the `tests/` directory separately like this:

```bash
eslint tests/ --no-eslintrc -c tests/.eslintrc.json
```

In the root `.eslintrc.json` you can add `"/tests/"` to `ignorePatterns` to prevent linting `tests/` with other commands (e.g., `eslint .`).
